src/test/datascience/kernel-launcher/localKernelFinder.unit.test.ts

```diff
@@ -994,3 +998,524 @@ import { disposeAllDisposables } from '../../../client/common/helpers';
         });
     });
 });
+
+[false, true].forEach((isWindows) => {
+    suite(`Local Kernel Finder ${isWindows ? 'Windows' : 'Unix'}`, () => {
+        let kernelFinder: ILocalKernelFinder;
+        let interpreterService: IInterpreterService;
+        let platformService: IPlatformService;
+        let fs: IFileSystem;
+        let extensionChecker: IPythonExtensionChecker;
+        const disposables: IDisposable[] = [];
+        let globalSpecPath: string;
+        type TestData = {
+            interpreters: {
+                interpreter: PythonEnvironment;
+                /**
+                 * These are all of the kernelspecs found within the Python environment.
+                 * Could be python or non-python kernlespecs.
+                 * Could be default or custom kernelspecs.
+                 */
+                kernelSpecs?: KernelSpec.ISpecModel[];
+            }[];
+            /**
+             * All of the globally installed KernelSpecs
+             */
+            globalKernelSpecs?: KernelSpec.ISpecModel[];
```

Comment:
Wrote some of the tests using a new way, wanted to be very explicit about the kernelspecs and the environments.
With the previous test (above this file), its very difficult to determine what kernlespec belongs where..

This type `TestData` is explicit in defining the global kernelspecs and where each kernel is located.
Need something as explicit as this for the tests, else its difficult to reason out the results and the failures.

Reply:
Also this way, we can mix and match kernelspecs now, 
E.g. i can create 2 default kernelspecs for 3 interpreters using the same objects.

Reply:
Should we just remove the old tests? if these are easier to write, why keep the other tests around?